### PR TITLE
Fix the API docs builder

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
Implemented a strategy for installing integrations in a way that doesn't break the API docs builder.
This also required a few additional changes:

1. no longer instantiating the ZenML store when the REST API utils module is being imported
2. using lazy initialization for the zen store in general, to avoid having to initializing the local sqlite zen store or having to connect to a zen server when not necessary (e.g. when running `zenml integration install`)
3. updating some integration package requirements:
  * kfp and kfp-tekton were out of sync with eachother
  * feast was using an older package version that was no longer working 

Tested that this works (i.e. it builds the API docs successfully in release branches) here: https://github.com/zenml-io/zenml/actions/runs/3245990909/jobs/5324213166

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/feast/__init__.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class FeastIntegration(Integration):
     """Definition of Feast integration for ZenML."""
 
     NAME = FEAST
-    REQUIREMENTS = ["feast[redis]>=0.19.4", "redis-server"]
+    REQUIREMENTS = ["feast[redis]~=0.26.0", "redis-server"]
```
Comment (strickvl) [inline src/zenml/integrations/feast/__init__.py:35]:
```suggestion
    REQUIREMENTS = ["feast[redis]~=0.26.0", "redis-server>=6.0.9"]
```

I made the same change in my branch working on fixing the Feast example, but added this. Hope pinning Redis doesn't mess with anything.

**Review by strickvl — CHANGES_REQUESTED** [review]
Looks great to me, but hope you can add in the `redis-server` change I suggested + that it doesn't break other things since I've added that same extended change elsewhere.

**Comment by strickvl** [comment]
Am certain @AlexejPenner will be happy about this PR.

**Review by AlexejPenner — APPROVED** [review]
Hallelujah, lgtm!

File: src/zenml/integrations/feast/__init__.py (hunk 1 above)
Comment (stefannica) [reply src/zenml/integrations/feast/__init__.py:35]: That shouldn't be a problem. I'll test it, just in case.

**Review by fa9r — APPROVED** [review]
Awesome, finally automated again 🙌

**Comment by stefannica** [comment]
New try to build the API docs with these changes and the GitHub action: https://github.com/zenml-io/zenml/actions/runs/3249084707/jobs/5331068077

File: src/zenml/zen_server/zen_server_api.py [hunk 2]
```diff
@@ -73,6 +77,16 @@ def relative_path(rel: str) -> str:
     allow_headers=["*"],
 )
 
+
+@app.on_event("startup")
+def initialize() -> None:
+    """Initialize the ZenML server."""
+    # IMPORTANT: this needs to be done before the fastapi app starts, to avoid
+    # race-conditions
```
Comment (strickvl) [inline src/zenml/zen_server/zen_server_api.py:85]:
```suggestion
    # race conditions
```

**Review by strickvl — APPROVED** [review]

**Pull request merged**